src/wx/vbam-options-static.cpp

```diff
@@ -0,0 +1,830 @@
+        { "Sound/GBAInterpolation", VbamOptionID::kSoundGBAInterpolation },
+        { "Sound/GBDeclicking", VbamOptionID::kSoundGBDeclicking },
+        { "Sound/GBEcho", VbamOptionID::kSoundGBEcho },
+        { "Sound/GBEnableEffects", VbamOptionID::kSoundGBEnableEffects },
+        { "Sound/GBStereo", VbamOptionID::kSoundGBStereo },
+        { "Sound/GBSurround", VbamOptionID::kSoundGBSurround },
+        { "Sound/Quality", VbamOptionID::kSoundQuality },
+        { "Sound/Volume", VbamOptionID::kSoundVolume },
+    };
+
+    const auto iter = kStringToOptionId.find(input);
+    if (iter == kStringToOptionId.end()) {
+        return nonstd::nullopt;
+    }
+    return iter->second;
+}
+
+wxString FilterToString(int value) {
+    assert(value >= 0 && static_cast<size_t>(value) < kNbFilterFunctions);
+    return kFilterStrings[value];
+}
+
+wxString InterframeToString(int value) {
+    assert(value >= 0 && static_cast<size_t>(value) < kNbInterframes);
+    return kInterframeStrings[value];
+}
+
+wxString RenderMethodToString(int value) {
+    assert(value >= 0 && static_cast<size_t>(value) < kNbRenderMethods);
+    return kRenderMethodStrings[value];
+}
+
+wxString AudioApiToString(int value) {
+    assert(value >= 0 && static_cast<size_t>(value) < kNbAudioApis);
+    return kAudioApiStrings[value];
+}
+
+wxString SoundQualityToString(int value) {
+    assert(value >= 0 && static_cast<size_t>(value) < kNbSoundQualities);
+    return kSoundQualityStrings[value];
+}
+
+int StringToFilter(const wxString& config_name, const wxString& input) {
+    static const std::map<wxString, FilterFunction> kStringToFilter = {
+        { kFilterStrings[0], FilterFunction::kNone },
+        { kFilterStrings[1], FilterFunction::k2xsai },
+        { kFilterStrings[2], FilterFunction::kSuper2xsai }, 
+        { kFilterStrings[3], FilterFunction::kSupereagle }, 
+        { kFilterStrings[4], FilterFunction::kPixelate }, 
+        { kFilterStrings[5], FilterFunction::kAdvmame }, 
+        { kFilterStrings[6], FilterFunction::kBilinear }, 
+        { kFilterStrings[7], FilterFunction::kBilinearplus }, 
+        { kFilterStrings[8], FilterFunction::kScanlines }, 
+        { kFilterStrings[9], FilterFunction::kTvmode }, 
+        { kFilterStrings[10], FilterFunction::kHQ2x }, 
+        { kFilterStrings[11], FilterFunction::kLQ2x }, 
+        { kFilterStrings[12], FilterFunction::kSimple2x }, 
+        { kFilterStrings[13], FilterFunction::kSimple3x }, 
+        { kFilterStrings[14], FilterFunction::kHQ3x }, 
+        { kFilterStrings[15], FilterFunction::kSimple4x }, 
+        { kFilterStrings[16], FilterFunction::kHQ4x }, 
+        { kFilterStrings[17], FilterFunction::kXbrz2x }, 
+        { kFilterStrings[18], FilterFunction::kXbrz3x }, 
+        { kFilterStrings[19], FilterFunction::kXbrz4x }, 
+        { kFilterStrings[20], FilterFunction::kXbrz5x }, 
+        { kFilterStrings[21], FilterFunction::kXbrz6x }, 
+        { kFilterStrings[22], FilterFunction::kPlugin }, 
```

Comment:
Any way to build this map iteratively?

Reply:
Yes, and that's actually better for long term maintenance, though we lose the const qualifier. Oh well, the one-time runtime cost is very low anyway.